framework/src/task_manager/task_manager_core.c

```diff
@@ -316,52 +338,83 @@ static int taskmgr_start(int handle, int caller_pid)
 	return OK;
 }
 
+void taskmgr_update_stop_status(int signo, siginfo_t *data)
+{
+	int fd;
+	int ret;
+	int handle;
+	handle = taskmgr_get_handle_by_pid(data->si_value.sival_int);
+
+	/* Terminate based on task type */
+	fd = taskmgr_get_drvfd();
+	if (fd < 0) {
+		tmdbg("Invalid driver fd\n");
+	}
+
+	ret = ioctl(fd, TMIOC_TERMINATE, TM_PID(handle));
+	if (ret == ERROR) {
+		tmdbg("Fail to terminate the task\n");
+	}
+
+	/* task or pthread terminated well */
+	TM_STATUS(handle) = TM_APP_STATE_STOP;
+
+	TM_FREE(TM_STOP_CB_INFO(handle));
+	TM_STOP_CB_INFO(handle) = NULL;
+}
+
 static int taskmgr_stop(int handle, int caller_pid)
 {
 	int ret;
+	union sigval msg;
+	tm_termination_info_t info;
 
 	if (IS_INVALID_HANDLE(handle)) {
 		return TM_INVALID_PARAM;
 	}
 
 	ret = taskmgr_get_task_state(handle);
-	if (ret == TM_APP_STATE_UNREGISTERED || ret == TM_APP_STATE_STOP) {
+	if (ret == TM_APP_STATE_UNREGISTERED || ret == TM_APP_STATE_STOP || ret == TM_APP_STATE_CANCELLING) {
+		if (ret == TM_APP_STATE_CANCELLING) {
+			ret = TM_APP_STATE_STOP;
```

Comment:
Do you need to change a status to STOP in this case? I think no problem on returning CANCELLING.